pkg/cmd/admin/prune/images.go

```diff
@@ -50,85 +60,138 @@ func NewCmdPruneImages(f *clientcmd.Factory, parentName, name string, out io.Wri
-				Streams:          allStreams,
-				Pods:             allPods,
-				RCs:              allRCs,
-				BCs:              allBCs,
-				Builds:           allBuilds,
-				DCs:              allDCs,
-				DryRun:           dryRun,
-				RegistryClient:   registryClient,
-				RegistryURL:      cfg.RegistryUrlOverride,
-			}
-			pruner := prune.NewImageRegistryPruner(options)
-
-			// this tabwriter is used by the describing*Pruners below for their output
-			w := tabwriter.NewWriter(out, 10, 4, 3, ' ', 0)
-			defer w.Flush()
-
-			imagePruner := &describingImagePruner{w: w}
-			imageStreamPruner := &describingImageStreamPruner{w: w}
-			layerPruner := &describingLayerPruner{w: w}
-			blobPruner := &describingBlobPruner{w: w}
-			manifestPruner := &describingManifestPruner{w: w}
-
-			switch cfg.Confirm {
-			case true:
-				imagePruner.delegate = prune.NewDeletingImagePruner(osClient.Images())
-				imageStreamPruner.delegate = prune.NewDeletingImageStreamPruner(osClient)
-				layerPruner.delegate = prune.NewDeletingLayerPruner()
-				blobPruner.delegate = prune.NewDeletingBlobPruner()
-				manifestPruner.delegate = prune.NewDeletingManifestPruner()
-			default:
-				fmt.Fprintln(os.Stderr, "Dry run enabled - no modifications will be made. Add --confirm to remove images")
-			}
+	allPods, err := kClient.Pods(kapi.NamespaceAll).List(labels.Everything(), fields.Everything())
+	if err != nil {
+		return err
+	}
 
-			err = pruner.Prune(imagePruner, imageStreamPruner, layerPruner, blobPruner, manifestPruner)
-			cmdutil.CheckErr(err)
-		},
+	allRCs, err := kClient.ReplicationControllers(kapi.NamespaceAll).List(labels.Everything())
+	if err != nil {
+		return err
 	}
 
-	cmd.Flags().BoolVar(&cfg.Confirm, "confirm", cfg.Confirm, "Specify that image pruning should proceed. Defaults to false, displaying what would be deleted but not actually deleting anything.")
-	cmd.Flags().DurationVar(&cfg.KeepYoungerThan, "keep-younger-than", cfg.KeepYoungerThan, "Specify the minimum age of a build for it to be considered a candidate for pruning.")
-	cmd.Flags().IntVar(&cfg.KeepTagRevisions, "keep-tag-revisions", cfg.KeepTagRevisions, "Specify the number of image revisions for a tag in an image stream that will be preserved.")
-	cmd.Flags().StringVar(&cfg.CABundle, "certificate-authority", cfg.CABundle, "The path to a certificate authority bundle to use when communicating with the managed Docker registries. Defaults to the certificate authority data from the current user's config file.")
-	cmd.Flags().StringVar(&cfg.RegistryUrlOverride, "registry-url", cfg.RegistryUrlOverride, "The address to use when contacting the registry, instead of using the default value. This is useful if you can't resolve or reach the registry (e.g.; the default is a cluster-internal URL) but you do have an alternative route that works.")
+	allBCs, err := osClient.BuildConfigs(kapi.NamespaceAll).List(labels.Everything(), fields.Everything())
+	err = oserrors.TolerateNotFoundError(err)
```

Comment:
Nvmd, I forgot about atomic part. On the other hand, this kind of plumbing will eventually bite us in the most unexpected way.